### PR TITLE
Add merkleblock and proof of work check

### PR DESCRIPTION
**Pull request by pebble8888**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.

### Description of the Change

Add merkleblock message validity check.
You should look out that the validity check here is not enough to insert to the local database.
Moreover you need timestamp check and etc.

1) Build merkle root from merkleblock message.
2) Check if calculated root match root hash in the message.
3) Check if block hash has enough proof of work.

UInt256 is for PoW difficulty check.
This implementation is large for it.

I have not enough time to amend this PR unfortunately.
If the implementation doesn't match your design, please reject it or pick up what you need.

### Alternate Designs

This check is implemented in BreadWallet.
I don't know we should check which message received.

### Benefits

Prevent interpolation and attack.

### Possible Drawbacks

Code maintainavility.
If a valid message is dropped, comment out the message check logic.
I tested some message is passed in test net.

### Applicable Issues

None

### CODE REVIEW

**Comment by akifuj** [comment]
Please modify copyright for each file, according to the existing files.

File: Sources/BitcoinKit/Core/ProofOfWork.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		= UInt256(data: Data(hex:"00000000ffffffffffffffffffffffffffffffffffffffffffffffffffffffff")!)!
+	static func isValidProofOfWork(blockHash: Data, bits: UInt32) -> Bool {
+		let target: UInt256
+		do {
+			target = try UInt256(compact: bits)
+		} catch {
+			// invalid bits
+			return false
+		}
+		if target == UInt256.zero {
```
Comment (akifuj) [inline Sources/BitcoinKit/Core/ProofOfWork.swift]: `guard` is better than  `if`.
This adapts to the following validations.

File: Sources/BitcoinKit/Networking/PeerGroup.swift [hunk 2]
```diff
@@ -103,6 +103,24 @@ public class PeerGroup: PeerDelegate {
     }
 
     public func peer(_ peer: Peer, didReceiveMerkleBlockMessage message: MerkleBlockMessage, hash: Data) {
+		if !ProofOfWork.isValidProofOfWork(blockHash: hash, bits: message.bits) {
```
Comment (akifuj) [inline Sources/BitcoinKit/Networking/PeerGroup.swift]: Please use `gaurd`

**Comment by pebble8888** [comment]
I fixed it for the code review comment.

**Comment by akifuj** [comment]
Thanks!

**Pull request closed without merging**